From: Christoph M. Becker <cmbecker69@gmx.de>
Date: Tue Jul 21 11:07:43 2020 +0200
Subject: [PATCH] Fix #79877: getimagesize function silently truncates after a null byte

We have to check for NUL bytes if `getimagesize()` has been called.

5.5 backport: https://github.com/shivammathur/php-src-backports/commit/022fda7dc238119d62c656ecd6ec4c16b4e3ab6d
---
diff --git a/ext/standard/image.c b/ext/standard/image.c
--- a/ext/standard/image.c
+++ b/ext/standard/image.c
@@ -1399,6 +1399,11 @@ static void php_getimagesize_from_any(INTERNAL_FUNCTION_PARAMETERS, int mode) {
 			return;
 	}
 
+	if (mode == FROM_PATH && CHECK_NULL_PATH(input, input_len)) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid path");
+		return;
+	}
+
 	if (argc == 2) {
 			zval_dtor(*info);
 			array_init(*info);
